### ISSUE TITLE
Bugs in D&D Beyond's implementation - 2: "Dice rolls not yet available for this section"

### ISSUE THREAD

**Issue by cbaer9** [skizzerz/DiceRoller#15]
Here's an example of a bug that surfaces on occasion in D&D Beyond's implementation of DiceRoller:

[https://www.dndbeyond.com/forums/d-d-beyond-general/play-by-post/54005-ghosts-of-saltmarsh-cbaer8-campaign-thread?comment=8565](https://www.dndbeyond.com/forums/d-d-beyond-general/play-by-post/54005-ghosts-of-saltmarsh-cbaer8-campaign-thread?comment=8565) 

As you can see, "Gerrard" makes a basic 1d20+5 initiative roll, but he gets a "Dice rolls not yet available for this section" result.

Two posts (and two hours) later, he's still not sure what the result is. He writes in post #8567, "Not available, then showed a 6, now not available again." So he rolls again.

Thanks again for any insights!

**Comment by cbaer9** [comment]
Here's a fresh example of this bug, from this morning. 

[https://www.dndbeyond.com/forums/d-d-beyond-general/play-by-post/54005-ghosts-of-saltmarsh-cbaer8-campaign-thread?comment=8735](https://www.dndbeyond.com/forums/d-d-beyond-general/play-by-post/54005-ghosts-of-saltmarsh-cbaer8-campaign-thread?comment=8735)

The player made a 1d20+x roll for perception, but received the "Dice rolls not yet available for this section" response.

Sometimes players will then edit the post, add a space after the roll, and then press return and cross their fingers; sometimes this works, sometimes it backfires and triggers a manipulation flag. (In this case the player chose not to, and instead rolled again using D&DB's sheet roller instead, which is recorded in the Game Log - never anyone's first choice in PBP.)

<img width="586" alt="Screenshot 2023-08-25 at 10 16 20 AM" src="https://github.com/skizzerz/DiceRoller/assets/17763215/6cf38089-865b-4796-a6e6-f6103e799135">

Thanks again for any insights!

**Comment by skizzerz** [comment]
I think this is largely the same as the previous issue you opened -- something in their forum software is not correctly mapping rolls to their associated post (instead grabbing the results from other posts, or posts that don't even exist yet). This would also explain the spurious cheat detection errors, if it's grabbing the rolls from someone else's post and then changing it to the rolls for your post.

tl;dr it's 100% on them and their forum implementation

**Comment by cbaer9** [comment]
Much obliged! I greatly appreciate the time you took to take a look. 

I will pass that along. I have little confidence that D&D Beyond will fix it anytime soon; it feels as if they have largely abandoned the play-by-post community in recent years, which is a shame. I hope I'm mistaken.

Thanks again for looking at the issues in their implementation of your code!
